mypy.ini

```diff
@@ -14,6 +14,7 @@ warn_unused_ignores = True
 warn_no_return = True
 warn_unreachable = True
 strict_equality = True
+enable_incomplete_feature = Unpack
```

Comment:
I noticed this was just marked stable in 1.7 that was just released, oddly coincidental timing! https://mypy-lang.blogspot.com/2023/11/mypy-17-released.html

But let's keep this as-is and upgrade mypy separately to limit the scope.